### PR TITLE
Add ability to set PEM content to Client Configuration

### PR DESCRIPTION
**Pull request by sergiishamrai-okta**
## Issue(s)

#486 
#488 (changes are based on this)

## Description

#488 was originally received from an external contributor. 

This PR along similar lines but uses a single `.setPrivateKey()` to take in either a PEM key file path or the PEM content as a string. Under the wraps, we detect if the user supplied a file or file content and handle the downstream accordingly.

## Category

- [ ] Bugfix
- [x] Enhancement
- [ ] New Feature
- [ ] Configuration Change
- [ ] Versioning Change
- [ ] Unit Test(s)
- [ ] Documentation

### CODE REVIEW

File: api/src/main/java/com/okta/sdk/client/ClientBuilder.java [hunk 1]
```diff
@@ -325,13 +328,50 @@
...
+     * Allows specifying the private key (PEM file) path (for private key jwt authentication) directly instead
+     * of relying on the default location + override/fallback behavior defined
+     * in the {@link ClientBuilder documentation above}.
+     *
+     * @param privateKeyStream representing an InputStream with private key PEM file content.
+     * @return the ClientBuilder instance for method chaining.
+     *
+     * @since 3.0.0
+     */
+    ClientBuilder setPrivateKey(InputStream privateKeyStream);
```
Comment (arvindkrishnakumar-okta) [inline api/src/main/java/com/okta/sdk/client/ClientBuilder.java:361]:
```suggestion
    ClientBuilder setPrivateKey(InputStream privateKeyInputStream);
```

File: README.md [hunk 2]
```diff
@@ -134,6 +134,10 @@ Client client = Clients.builder()
     .setClientId("{clientId}")
     .setScopes(new HashSet<>(Arrays.asList("okta.users.read", "okta.apps.read")))
     .setPrivateKey("/path/to/yourPrivateKey.pem")
+    // (or) .setPrivateKey("full PEM payload")
+    // (or) .setPrivateKey(Paths.get("/path/to/yourPrivateKey.pem"))
+    // (or) .setPrivateKey(inputStream)
```
Comment (arvindkrishnakumar-okta) [inline README.md:139]: Supporting both `.setPrivateKey("full PEM payload")` & `.setPrivateKey(inputStream)` seems unnecessary to me. `.setPrivateKey("full PEM payload")` should be enough between the two I think.

File: api/src/main/java/com/okta/sdk/client/ClientBuilder.java [hunk 3]
```diff
@@ -325,13 +328,50 @@
...
+     * @since 3.0.0
+     */
+    ClientBuilder setPrivateKey(Path privateKeyPath);
+
+    /**
+     * Allows specifying the private key (PEM file) path (for private key jwt authentication) directly instead
+     * of relying on the default location + override/fallback behavior defined
+     * in the {@link ClientBuilder documentation above}.
+     *
+     * @param privateKeyStream representing an InputStream with private key PEM file content.
```
Comment (arvindkrishnakumar-okta) [inline api/src/main/java/com/okta/sdk/client/ClientBuilder.java:356]:
```suggestion
     * @param privateKeyInputStream representing an InputStream with private key PEM file content.
```

File: api/src/main/java/com/okta/sdk/client/ClientBuilder.java [hunk 4]
```diff
@@ -325,13 +328,50 @@
...
+     * @since 3.0.0
+     */
+    ClientBuilder setPrivateKey(InputStream privateKeyStream);
+
+    /**
+     * Allows specifying the private key (PEM file) path (for private key jwt authentication) directly instead
+     * of relying on the default location + override/fallback behavior defined
+     * in the {@link ClientBuilder documentation above}.
+     *
+     * @param privateKey representing an PrivateKey PEM file.
```
Comment (arvindkrishnakumar-okta) [inline api/src/main/java/com/okta/sdk/client/ClientBuilder.java:368]:
```suggestion
     * @param privateKey the {@link PrivateKey} instance.
```

Comment (arvindkrishnakumar-okta) [reply api/src/main/java/com/okta/sdk/client/ClientBuilder.java:368]: `java.security.PrivateKey` instance.

File: examples/quickstart/src/main/java/quickstart/ReadmeSnippets.java [hunk 5]
```diff
@@ -74,6 +74,10 @@ private void createOAuth2Client() {
             .setClientId("{clientId}")
             .setScopes(new HashSet<>(Arrays.asList("okta.users.read", "okta.apps.read")))
             .setPrivateKey("/path/to/yourPrivateKey.pem")
+            // (or) .setPrivateKey("full PEM payload");
+            // (or) .setPrivateKey(Paths.get("/path/to/yourPrivateKey.pem"));
+            // (or) .setPrivateKey(inputStream);
```
Comment (arvindkrishnakumar-okta) [inline examples/quickstart/src/main/java/quickstart/ReadmeSnippets.java:79]: see my earlier comment on `.setPrivateKey(inputStream)`.

File: impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java [hunk 6]
```diff
@@ -391,6 +403,70 @@ public ClientBuilder setPrivateKey(String privateKey) {
...
+            this.clientConfig.setPrivateKey(getFileContent(privateKeyStream));
+        }
+        return this;
+    }
+
+    @Override
+    public ClientBuilder setPrivateKey(PrivateKey privateKey) {
+        if (isOAuth2Flow()) {
+            Assert.notNull(privateKey, "Missing privateKeyFile");
+            String encodedString = "-----BEGIN PRIVATE KEY-----\n";
```
Comment (arvindkrishnakumar-okta) [inline impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java:428]: RSA private key usually has different headers `-----BEGIN RSA PRIVATE KEY-----` & `-----END RSA PRIVATE KEY` . Please check if this works with other key types we support (RSA, EC etc).

File: README.md (hunk 2 above)
Comment (sergiishamrai-okta) [reply README.md:139]: @arvindkrishnakumar-okta  Do you want to remove it from README.md only? Or you want to completely remove implementation starting from defining this method in the ClientBuilder interface?

File: api/src/main/java/com/okta/sdk/client/ClientBuilder.java (hunk 4 above)
Comment (sergiishamrai-okta) [reply api/src/main/java/com/okta/sdk/client/ClientBuilder.java]: fixed

File: api/src/main/java/com/okta/sdk/client/ClientBuilder.java (hunk 3 above)
Comment (sergiishamrai-okta) [reply api/src/main/java/com/okta/sdk/client/ClientBuilder.java]: fixed

File: api/src/main/java/com/okta/sdk/client/ClientBuilder.java (hunk 1 above)
Comment (sergiishamrai-okta) [reply api/src/main/java/com/okta/sdk/client/ClientBuilder.java]: fixed

File: impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java (hunk 6 above)
Comment (sergiishamrai-okta) [reply impl/src/main/java/com/okta/sdk/impl/client/DefaultClientBuilder.java]: Updated

File: README.md (hunk 2 above)
Comment (sergiishamrai-okta) [reply README.md:139]: Implemented in here https://github.com/okta/okta-sdk-java/pull/498

Comment (arvindkrishnakumar-okta) [reply README.md:139]: sorry, rethinking this I think we should retain `.setPrivateKey(inputStream)`.

**Review by arvindkrishnakumar-okta — APPROVED** [review]

**Comment by arvindkrishnakumar-okta** [comment]
Please update https://github.com/okta/okta-sdk-java#yaml-configuration section.

**Pull request merged**